x-pack/plugins/maps/public/shared/layers/vector_layer.js

```diff
@@ -527,7 +527,7 @@ export class VectorLayer extends AbstractLayer {
   }
 
   canShowTooltip() {
-    return this._source.canFormatFeatureProperties();
+    return this.isVisible() && this._source.canFormatFeatureProperties();
```

Comment:
small optimization and unrelated to this bug fix.

This doesn't change the behavior in practice because the corresponding mapbox-layers are turned invisible as well, which would not return any results to begin with.